SQL/New_patches/2018-09-07_EEG_tables.sql

```diff
@@ -0,0 +1,259 @@
+CREATE TABLE `physiological_status_type` (
+  `PhysiologicalStatusTypeID` TINYINT(2) UNSIGNED NOT NULL AUTO_INCREMENT,
+  `ChannelStatus`             VARCHAR(10)         NOT NULL,
+  PRIMARY KEY (`PhysiologicalStatusTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+-- Create a physiological_channel_type table
+CREATE TABLE `physiological_channel_type` (
+  `PhysiologicalChannelTypeID` TINYINT(3) UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `ChannelType`                VARCHAR(255)        NOT NULL,
+  `ChannelDescription`         VARCHAR(255)        DEFAULT NULL,
+  PRIMARY KEY (`PhysiologicalChannelTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+-- Create a physiological_channel table that will store all information from
+CREATE TABLE `physiological_channel` (
+  `PhysiologicalChannelID`     INT(10) UNSIGNED    NOT NULL      AUTO_INCREMENT,
+  `PhysiologicalFileID`        INT(10) UNSIGNED    NOT NULL,
+  `PhysiologicalChannelTypeID` TINYINT(3) UNSIGNED NOT NULL,
+  `PhysiologicalStatusTypeID`  TINYINT(2) UNSIGNED DEFAULT NULL,
+  `Name`                       VARCHAR(50)         NOT NULL,
+  `Description`                VARCHAR(255)        DEFAULT NULL,
+  `SamplingFrequency`          INT(6)              DEFAULT NULL,
+  `LowCutoff`                  DECIMAL(8,3)        DEFAULT NULL,
+  `HighCutoff`                 DECIMAL(8,3)        DEFAULT NULL,
+  `ManualFlag`                 DECIMAL(5,4)        DEFAULT NULL,
+  `Notch`                      INT(6)              DEFAULT NULL,
+  `StatusDescription`          VARCHAR(255)        DEFAULT NULL,
+  `Unit`                       VARCHAR(255)        DEFAULT NULL,
+  `File`                       VARCHAR(255)        DEFAULT NULL,
+  `SoftwareFilters`            VARCHAR(255)        DEFAULT NULL,
+  PRIMARY KEY (`PhysiologicalChannelID`),
+  CONSTRAINT `FK_physiological_file_PhysiologicalFileID_2`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE,
+  CONSTRAINT `FK_physiological_channel_type_PhysiologicalChannelTypeID`
+    FOREIGN KEY (`PhysiologicalChannelTypeID`)
+    REFERENCES `physiological_channel_type` (`PhysiologicalChannelTypeID`),
+  CONSTRAINT `FK_physiological_status_type_PhysiologicalStatusTypeID`
+    FOREIGN KEY (`PhysiologicalStatusTypeID`)
+    REFERENCES `physiological_status_type` (`PhysiologicalStatusTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+
+-- Create a physiological_electrode table
+CREATE TABLE `physiological_electrode` (
+  `PhysiologicalElectrodeID` INT(10) UNSIGNED NOT NULL      AUTO_INCREMENT,
+  `PhysiologicalFileID`      INT(10) UNSIGNED NOT NULL,
+  `Name`                     VARCHAR(50)      NOT NULL,
+  `X`                        DECIMAL(12,6)    NOT NULL,
+  `Y`                        DECIMAL(12,6)    NOT NULL,
+  `Z`                        DECIMAL(12,6)    NOT NULL,
+  `Type`                     VARCHAR(50)      NOT NULL,
+  `Material`                 VARCHAR(50)      NOT NULL,
```

Comment:
Should the Type and Material attributes be in a separate lookup table?

Reply:
mmm... Good point. The problem is that I have no idea what should go in those lookup tables as it is not specified in the BIDS spec. Maybe I could indeed have look up tables and insert into them each time there is something new for type and material with the insert scripts? @PapillonMcGill What do you think?

Reply:
Can we ask an EEG expert or do a little search? I would like to know more. Specially if a single Type always have the same Material. It might also be very limited but new type show up once in a while when manufacturer introduce them. In that case a lookup table that simply add a missing value on the go

Reply:
OK. I will contact the EEG people I know and let you know what they say. Thanks!

Reply:
@PapillonMcGill I had news regarding these data types from the EEG experts.

"There is no official names for type and materials because manufacturers can change as they see fit."

Based on that reply, I could either create two look up tables (one for type and one for material) and insert in them each time the insertion script encounter a new type (or new material) or leave it to VARCHAR. Let me know what you think.

I also asked them questions about new fields that are present in the BIDS structure and will add them in the next commit.

Reply:
As theses 2 parameters seem to be at manufacturer discretion and new type and material can appear as they introduce new products, they have to be open for additional value on the fly.
But they will probably be from a limited list, thus repeating values.

Having a reference table with a pre-defined list from which to take value as long as the code let the user add new values. 
This would ensure consistency in name and would facilitate adding to a list of predefined value without having to modify the code with each new addition.

Reply:
Sure! I can implement that. Will be doing that today.